File: application/app/src/main/java/com/yahyeet/boardbook/model/mock/repository/MockUserRepository.java
package com.yahyeet.boardbook.model.mock.repository;

import com.yahyeet.boardbook.model.entity.User;
import com.yahyeet.boardbook.model.repository.IUserRepository;

import java.util.List;
import java.util.concurrent.CompletableFuture;

/**
 * Mock implementation of a user repository
 */
public class MockUserRepository extends AbstractMockRepository<User> implements IUserRepository {
	@Override
	public CompletableFuture<List<User>> findFriendsByUserId(String id) {
		return find(id).thenApply(User::getFriends);
	}
}

File: application/app/src/main/java/com/yahyeet/boardbook/model/mock/repository/MockUserRepository.kt
package com.yahyeet.boardbook.model.mock.repository

import com.yahyeet.boardbook.model.entity.Match
import com.yahyeet.boardbook.model.entity.User
import com.yahyeet.boardbook.model.repository.IRepositoryListener
import com.yahyeet.boardbook.model.repository.IUserRepository

import java.util.ArrayList
import java.util.HashMap
import java.util.concurrent.CompletableFuture
import java.util.concurrent.CompletionException
import java.util.concurrent.atomic.AtomicBoolean
import java.util.function.Function

class MockUserRepository : AbstractMockRepository<User>(), IUserRepository {

    override fun findFriendsByUserId(id: String): CompletableFuture<List<User>> {
        return find(id).thenApply { user -> user.friends }
    }
}
